lib/ae_mdw/db/util.ex

```diff
@@ -50,6 +53,19 @@ defmodule AeMdw.Db.Util do
     end
   end
 
+  @spec last_gen_and_time(state()) :: {Blocks.height(), Blocks.time()}
+  def last_gen_and_time(state) do
+    case State.prev(state, Model.Block, nil) do
+      {:ok, {height, _mbi}} ->
+        Model.block(hash: block_hash) = State.fetch!(state, Model.Block, {height, -1})
+
+        {height, block_time(block_hash)}
```

Comment:
This would cause to retrieve the time for the last block multiple times when requesting it for a list of oracles: https://github.com/aeternity/ae_mdw/pull/1390/files#diff-d800a239402e86baa72793fe422de5f1d247ad3452b2efd3ee981cede91c038eR298